src/pages/ServiceDetails/ServiceDetailsPage.tsx

```diff
@@ -93,14 +93,13 @@ class ServiceDetails extends React.Component<ServiceDetailsProps, ServiceDetails
             namespaces.map(ns => API.getIstioConfig(ns.name, ['gateways'], false, '', ''))
           )
           .then(responses => {
-            let gatewayList: string[] = [];
+            let gateways: Gateway[] = [];
             responses.forEach(response => {
-              const ns = response.data.namespace;
               response.data.gateways.forEach(gw => {
-                gatewayList = gatewayList.concat(ns.name + '/' + gw.metadata.name);
+                gateways.push(gw);
               });
             });
-            this.setState({ gateways: gatewayList });
+            this.setState({ gateways: gateways });
```

Comment:
You probably can simplify lines 96-102 to a single statement:

```js
this.setState({ gateways: responses.map(response => response.data.gateways).flat() });
```

Reply:
Yes, sure, thanks!.